### ISSUE TITLE
pyhf loading `histosys` modifier for `normsys` definition

### ISSUE THREAD

**Issue by kratsg** [diana-hep/pyhf#208]
# Description

Filing this to look at in a little while. I've got a local JSON of MBJ from an exclusion fit multibin, Gtt, that is validated against the schema. When inflating the `Model`, I get a crash. 

Clearly, there's some sort of mismatch where `histosys` is being given a `normsys` definition, or we're loading `histosys` for a `normsys` definition. Not sure where this mismatch occurs but will dig into it soon.

# Expected Behavior

`Model` is inflated correctly.

# Actual Behavior

A crash occurs.

```python
>>> import pyhf
>>> import json
>>> d = json.load(file('3b_tag21.2.27-1_RW_ExpSyst_79800_multibin_excl_Gtt_2400_5000_800.json'))
>>> pyhf.utils.validate({'channels': d['channels']}, pyhf.utils.get_default_schema())
>>> try:
...   pdf = pyhf.Model({'channels': d['channels']}, poiname='mu_SIG')
... except:
...   import traceback, sys, code
...   type, value, tb = sys.exc_info()
...   traceback.print_exc()
...   last_frame = lambda tb=tb: last_frame(tb.tb_next) if tb.tb_next else tb
...   frame = last_frame().tb_frame
...   ns = dict(frame.f_globals)
...   ns.update(frame.f_locals)
...   code.interact(local=ns)
... 

Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/kratsg/pyhf/pyhf/pdf.py", line 106, in __init__
    self.config = _ModelConfig.from_spec(spec,**config_kwargs)
  File "/Users/kratsg/pyhf/pyhf/pdf.py", line 19, in from_spec
    modifier.add_sample(channel, sample, modifier_def)
  File "/Users/kratsg/pyhf/pyhf/modifiers/histosys.py", line 34, in add_sample
    self.at_minus_one.setdefault(channel['name'], {})[sample['name']] = modifier_def['data']['lo_data']
KeyError: 'lo_data'
Python 2.7.14 (default, Mar 22 2018, 15:04:47) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> self
<pyhf.modifiers.histosys.histosys object at 0x119392c90>
>>> modifier_def
{u'type': u'normsys', u'data': {u'lo': 0.953548, u'hi': 1.04867}, u'name': u'RW_1CR'}
```

# Steps to Reproduce

Detail your environment and the commands that you are executing to generate the bug. I am using `master` of `pyhf` at ee1025037acad4b36e5c9d62402c00119e6c6cb5.

The JSON is available in this [gist](https://gist.github.com/21655b798591c8e85d6262592eeb0a7a).

# Checklist

- [x] Run `git fetch` to get the most up to date version of `master`
- [x] Searched through existing Issues to confirm this is not a duplicate issue
- [x] Filled out the Description, Expected Behavior, Actual Behavior, and Steps to Reproduce sections above or have edited/removed them in a way that fully describes the issue

**Comment by lukasheinrich** [comment]
I suspect this is because the same name is used for multiple modifiers of different type. will confirm in a  bit

**Comment by lukasheinrich** [comment]
yeah that's it. I'll PR

**Comment by kratsg** [comment]
> I suspect this is because the same name is used for multiple modifiers of different type. will confirm in a bit

Damn. I didn't catch that. Nice.

**Comment by lukasheinrich** [comment]
resolved by #209
